### PR TITLE
test(websocket-js): add snapshot test for CompileOperationSchemas component

### LINKED ISSUE #1935: [TEST] Add component tests for CompileOperationSchemas.js (Js WebSocket Template)
Description
Adds unit tests for the CompileOperationSchemas.js` component in the Javascript WebSocket client template. The component contains complex conditional logic but currently has no component-level test coverage.

Component location
`packages/templates/clients/websocket/javascript/components/CompileOperationSchemas.js`

I’m happy to submit a PR implementing this.

### PR DESCRIPTION
**Pull request by SHUBHANSHU602**
###  Summary

This PR introduces unit tests for the `CompileOperationSchemas` component in the WebSocket JavaScript client template.

The component is responsible for generating a method (`compileOperationSchemas`) as a string using the `Text` component from `@asyncapi/generator-react-sdk`.

---

###  What is tested

* Returns `null` when:

  * `sendOperations` is `undefined`
  * `sendOperations` is an empty array

* Renders a `Text` component when `sendOperations` are provided

* Verifies that the generated output string contains key parts:

  * `async compileOperationSchemas()`
  * `this.schemasCompiled`
  * `compileSchemasByOperationId`

---
Screenshot 2026-03-28 122237

Resolves #1935

## Summary by CodeRabbit

* **Tests**
  * Added tests for the CompileOperationSchemas component covering normal rendering and edge cases (empty, null, or missing input), ensuring stable output and empty output where appropriate.

### CODE REVIEW

**Review by Adi-204 — COMMENTED** [review]
@SHUBHANSHU602 the tests are incorrect. First, you need to create a components folder inside the test directory to match the expected structure. Also, we use snapshot testing for components—not unit tests. I’d suggest looking at other templates, such as the Java template, which already has proper component testing in place.
https://github.com/asyncapi/generator/tree/master/packages/templates/clients/websocket/java/quarkus/test/components

**Comment by SHUBHANSHU602** [comment]
ohk @Adi-204 I was bit confused about the directory , thanks for guidence I will fix it soon.

**Comment by SHUBHANSHU602** [comment]
@Adi-204 let me know if it is ohk now

File: packages/templates/clients/websocket/javascript/test/CompileOperationSchemas.test.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  it('should contain compileOperationSchemas function in output', () => {
+    const result = CompileOperationSchemas({ sendOperations: ['op1'] });
+
+    const textContent = result.props.children.props.children;
+
+    expect(textContent).toContain('async compileOperationSchemas()');
+    expect(textContent).toContain('this.schemasCompiled');
+    expect(textContent).toContain('compileSchemasByOperationId');
+  });
+});
```
Comment (Adi-204) [inline packages/templates/clients/websocket/javascript/test/CompileOperationSchemas.test.js]: please delete this test file as we don't need it anymore.

File: packages/templates/clients/websocket/javascript/test/components/CompileOperationSchemas.test.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  });
+
+  it('renders empty output when sendOperations is undefined', async () => {
+    const result = await render(
+      <CompileOperationSchemas />
+    );
+
+    expect(result).toBe('');
+  });
+});
```
Comment (Adi-204) [inline packages/templates/clients/websocket/javascript/test/components/CompileOperationSchemas.test.js:54]: just add one more testcase with sendOperations=null

**Review by Adi-204 — COMMENTED** [review]
@SHUBHANSHU602 left 2 comments rest looks good!

File: packages/templates/clients/websocket/javascript/test/components/CompileOperationSchemas.test.js (hunk 2 above)
Comment (SHUBHANSHU602) [reply packages/templates/clients/websocket/javascript/test/components/CompileOperationSchemas.test.js:35]: @Adi-204 I have added the required testcase

File: packages/templates/clients/websocket/javascript/test/components/CompileOperationSchemas.test.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import { CompileOperationSchemas } from '../../components/CompileOperationSchemas';
+
+describe('CompileOperationSchemas', () => {
+  it('renders correctly with send operations', async () => {
+    const result = await render(
+      <CompileOperationSchemas sendOperations={['testOperation']} />
+    );
+
+    expect(result).toMatchSnapshot();
+  });
```
Comment (Adi-204) [inline packages/templates/clients/websocket/javascript/test/components/CompileOperationSchemas.test.js:11]: This is not wrong but we can do better by extracting send operations from asyncapi doc as same is done in component basically just do what is done in component https://github.com/asyncapi/generator/blob/3c012395db913803d8fcec949bf7a4d034a7aae2/packages/templates/clients/websocket/python/test/components/Send.test.js
you also need to use same asyncapiFilePath `asyncapi-websocket-components.yml` for testing.

**Review by Adi-204 — COMMENTED** [review]
@SHUBHANSHU602 left a comment!

File: packages/templates/clients/websocket/javascript/test/components/CompileOperationSchemas.test.js (hunk 3 above)
Comment (SHUBHANSHU602) [reply packages/templates/clients/websocket/javascript/test/components/CompileOperationSchemas.test.js:11]: ohk @Adi-204 I will fix it

**Comment by derberg** [comment]
@SHUBHANSHU602 do you continue with this one?

**Comment by SHUBHANSHU602** [comment]
@derberg I am having my exams till 25 April, After that I will continue, If there is any kind of time limit ,you please close my PR  and  ask any one else to continue 💜 Thanks

**Comment by SHUBHANSHU602** [comment]
@coderabbitai , does my now satisfy the need of issue , also consider this statement by maintainer ,"This is not wrong but we can do better by extracting send operations from asyncapi doc as same is done in component basically just do what is done in component https://github.com/asyncapi/generator/blob/3c012395db913803d8fcec949bf7a4d034a7aae2/packages/templates/clients/websocket/python/test/components/Send.test.js
you also need to use same asyncapiFilePath asyncapi-websocket-components.yml for testing"

**Comment by SHUBHANSHU602** [comment]
@derberg @Adi-204 Please have a look now.

**Review by Adi-204 — APPROVED** [review]

**Comment by SHUBHANSHU602** [comment]
Thanks
